ksql-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/KsqlMaterializationFactory.java

```diff
@@ -219,10 +168,60 @@ SqlPredicate create(
 
     KsqlMaterialization create(
         Materialization inner,
-        Function<GenericRow, GenericRow> aggregateTransform,
-        Predicate<Struct, GenericRow> havingPredicate,
-        Function<GenericRow, GenericRow> storeToTableTransform,
-        LogicalSchema schema
+        LogicalSchema schema,
+        List<BiFunction<Struct, GenericRow, Optional<GenericRow>>> transforms
     );
   }
+
+  private class TransformVisitor implements MaterializationInfo.TransformVisitor<
+      BiFunction<Struct, GenericRow, Optional<GenericRow>>> {
+    private final QueryContext.Stacker stacker;
+
+    TransformVisitor(final QueryContext.Stacker stacker) {
+      this.stacker = Objects.requireNonNull(stacker, "stacker");
+    }
+
+    @Override
+    public BiFunction<Struct, GenericRow, Optional<GenericRow>> visit(
+        final MaterializationInfo.AggregateMapInfo info) {
+      final Function<GenericRow, GenericRow> mapper = aggregateMapperFactory.create(
+          info.getInfo(),
+          functionRegistry
+      );
+      return (s, g) -> Optional.of(mapper.apply(g));
```

Comment:
nit: `s` and `g` are the key and value, which would more normally be represented with `k` and `v`, i.e. name them after what they are, not what type they are.